src/hooks/useClusters.tsx

```diff
@@ -1,12 +1,50 @@
-import { useQuery } from 'react-query';
+import { useMutation, useQuery, useQueryClient } from 'react-query';
 
 import { Queries } from '@/lib/react-query';
 import { REGION_ENVS } from '@/services/BaseService';
-import { fetchJobClusters } from '@/services/JobClusterService';
+import {
+  fetchJobClusterByName,
+  fetchJobClusters,
+  fetchJobsOnCluster,
+} from '@/services/JobClusterService';
+import { killJobs } from '@/services/JobService';
+import { showErrorNotification, showSuccessNotification } from '@/utils/notifications';
 
 export function useClusters() {
   return useQuery({
     queryKey: [Queries.CLUSTERS],
     queryFn: () => fetchJobClusters(REGION_ENVS),
   });
 }
+
+export function useClusterDetails(clusterName: string) {
+  const shouldFetch = Boolean(clusterName);
+  return useQuery({
+    queryKey: [Queries.CLUSTER_DETAILS, clusterName],
+    queryFn: () => fetchJobClusterByName(REGION_ENVS, clusterName),
+    enabled: shouldFetch,
+  });
+}
+
+export function useJobsOnCluster(clusterName: string) {
+  const shouldFetch = Boolean(clusterName);
+  return useQuery({
+    queryKey: [Queries.JOBS_ON_CLUSTER, clusterName],
```

Comment:
Maybe make the queryKey: [Queries. CLUSTER_DETAILS, clusterName, Queries.JOBS_ON_CLUSTER]

so its more hierarchical. That way if you invalidate, Queries.CLUSTER_DETAILS, for example, all children queries automatically get invalidated too.